crates/proto/src/rr/rdata/txt.rs

```diff
@@ -59,6 +59,26 @@ impl TXT {
         }
     }
 
+    /// Creates a new TXT record data from bytes.
+    /// Allows creating binary record data.
+    ///
+    /// # Arguments
+    ///
+    /// * `txt_data` - the set of bytes which make up the txt_data.
+    ///
+    /// # Return value
+    ///
+    /// The new TXT record data.
+    pub fn from_bytes(txt_data: Vec<&[u8]>) -> TXT {
```

Comment:
this would be a little more flexible as a `pub fn from_bytes(txt_data: &[&[u8]]) -> TXT`, would it not? We're not using the ownership of the Vec in this case, so seems liek it might be better to take it as a ref?

Reply:
I've considered that first, and I've moved into an owned Vec after a couple of things:

- It matches the `Vec<String>` method, taking a `Vec<_>` of values.
- When writing the test, it got a bit cumbersome doing `&vec![&vec![1, 2, 3]]`, as it [cannot infer the derefs appropriately](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=bc7dab4434aa280c52c04caa8f78c4a9), and putting it on intermediate variables to read better.
- Even on separate variables, [it requires type hintings](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e645f165d282757baa96504e2b698535)
- It requires an extra `.to_vec()` before `.into_iter()`, maybe it is double allocating if someone provides a fresh `vec![]` as argument?

Reply:
The original method of `Vec<String>` could get optimized out by the compiler. The goal here is to make the cost obvious to the caller, so that they know that this is going to allocate. That's at least the thought.

What do you think of this interface `pub fn from_bytes<B: Into<Box<[u8]>>>(txt_data: Vec<B>) -> TXT`?

see here: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ee3e2c3a456ddca4ef3082261b86c9de

Reply:
I think it does provide more context that it would allocate with this signature. The trade-off is requiring more manual type hinting on some cases as displayed on the playground.

Would you like me to change the PR to that signature and send it again?

Reply:
I see your point about the type hinting. Not sure if there’s a better signature that might help with that. If you think that interface is too cumbersome, then I’m ok with what you have here.

If we don’t want to make the change, then we can merge as is.